### ISSUE TITLE
CompoundSpectralModel Norm issue

### ISSUE THREAD

**Issue by MRegeard** [gammapy/gammapy#4008]
**Gammapy version**
Gammapy 0.20.1

**Bug description**
It appears that while fixing the bug related to FluxPoints when using a CompoundSectralModel, it is now (Gammapy-0.20.1) impossible to create a CompoundSpectralModel with more than one free amplitude in the spectral models. 

**Expected behavior**
I think that it should not be the case, at least for the fitting of the data. In Gammapy-0.19, this was possible, however, it was not possible to compute FluxPoints.
Of course, in the case of degenarated models, amplitude should be fixed, but I think that it is the responsibility of the user rather than an obligation for the usage of CompoundSpectralModel. 
But then, it raise the issue of how do we compute FluxPoints with multiple free amplitude in CompoundSpectralModel. 

**Temporary fix**
A possible fix to this issue is to set _is_norm to False to each amplitude in the CompoundSpectralModel but one. However, this is_norm feature does not seems very clear (at least to me), and should probably be define in an other way.

**Comment by adonath** [comment]
Thanks @MRegeard!

In the Gammapy dev meeting we agreed that it would be the desired behavior to check for the uniqueness of the free amplitude parameter. This is done here: https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/cube.py#L86. The reason is that two amplitude parameters will be correlated 100% in a fit and will never yield stable results. I don't understand where the limitation to a single free parameter leads to undesired behavior in your case.

**Comment by adonath** [comment]
>  how do we compute FluxPoints with multiple free amplitude in CompoundSpectralModel.

The answer here is that this is unfeasible and should never be done. The two different amplitude parameters will be correlated 100%. In case you have done this before, you should not trust those previous results and rely on the new behavior instead.

**Comment by registerrier** [comment]
I think the issue here is that we overlooked that some models can have independent and not correlated amplitudes: e.g. PL + gaussian line  or if you do a fit over a sufficiently broad range you can have several PLs whose amplitudes are controlled by different datasets (e.g. Fermi + VHE). 

This type of use case is no longer supported, in particular in the absence of a setter for `Parameter.is_norm`.

**Comment by registerrier** [comment]
> So the current check is strictly speaking only valid if the operator is multiplicative?

Yes I would think so, what we want to avoid is having a model with a free norm multiplying a model with a free amplitude.

**Comment by adonath** [comment]
The action item here for v1.0 is to additionally check the operator to be multiplicative...

**Comment by bkhelifi** [comment]
If I understand correctly, the proposal is to remove  https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/cube.py#L86 AND to check in the FluxPointEstimator if the spectral model has several free norm??

**Comment by adonath** [comment]
@bkhelifi No, https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/cube.py#L86 can remain, however we should check whether the operator is multiplicative in addition. Does this make sense?

**Comment by registerrier** [comment]
What is there left to be done here after #4040 ?

It is now possible to have a `CompoundSpectralModel` with two free `is_norm` parameters. Currently the `FluxEstimator` takes the first `Parameter` which has `is_norm == True` as the scaled parameter. Probably one should check that the parameter is not frozen too?
https://github.com/gammapy/gammapy/blob/6c0fc3ad452cb34e01161de4892714096f2bfcf1/gammapy/estimators/flux.py#L117

**Comment by bkhelifi** [comment]
In the `FluxEstimator`, the first parameter is always the one of the `ScaleSpectralModel` that is created internally by the estimator, the parameters of the other models (in particular the ones of `CompoundSpectralModel` if any) are frozen.
Is it really necessary that this non-accessible `ScaleSpectralModel` created correctly should be checked? I can add this check per safety... As you prefer.

**Comment by registerrier** [comment]
> In the FluxEstimator, the first parameter is always the one of the ScaleSpectralModel that is created internally by the estimator, the parameters of the other models (in particular the ones of CompoundSpectralModel if any) are frozen. Is it really necessary that this non-accessible ScaleSpectralModel created correctly should be checked? I can add this check per safety... As you prefer.

Not really. The `scale_parameter` is taken from the `reference_model`. It is used to determine the allowed range for the `norm` of the `ScaleSpectralModel`. See:
https://github.com/gammapy/gammapy/blob/6c0fc3ad452cb34e01161de4892714096f2bfcf1/gammapy/estimators/flux.py#L83

This is the mechanism we introduced to pass user choices for the amplitude range (e.g. positive or not) to the `FluxEstimator`. Now, with two `is_norm` parameters if the reference model is a `CompoundSpectralModel`, we don't know which one will be used a priori since it depends on how the compound is built. This might be problematic.

An option could be to check that the parameter is also free. This would require the user to select which model he/she wants to use as reference.

**Comment by bkhelifi** [comment]
So, the idea is to create a new PR for `FluxEstimator.get_scale_model` containing a check on the number of non-frozen 'norm' parameters of the reference model is greater to 1... Is it correct?
